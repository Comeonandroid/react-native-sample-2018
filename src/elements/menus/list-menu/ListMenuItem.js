import React, { PureComponent } from 'react'
import styled from 'styled-components'
import { PropTypes } from 'prop-types'

import { View, Text, TouchableHighlight } from '../../index'
import Icon from '../../../utils/icon'

class ListMenuItem extends PureComponent {

  constructor(params) {
    super(params)

    this.handleClick = this.handleClick.bind(this)
  }

  handleClick() {
    const { name, onPress } = this.props
    if(onPress) { onPress(name) }
  }

  render() {
    const { text, children, separator, className } = this.props
    return(
      <TouchableHighlight onPress={ this.handleClick }>
        <Wrapper
          separator={ separator }
          className={ className }>

          <Label>{ text }</Label>
          { children || <Icon name='right-arrow' size={12}></Icon> }
        </Wrapper>
      </TouchableHighlight>
    )
  }
}

ListMenuItem.propTypes = {
  children: PropTypes.node,
  name: PropTypes.string,
  text: PropTypes.string,
  className: PropTypes.string,
  separator: PropTypes.bool,
  onPress: PropTypes.func
}

const Wrapper = styled(View) `
  height: 45px;
  padding: 0px 20px 0px 16px;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  background-color: ${ ({ theme }) => theme.colors.backgroundPage };
  borderBottomWidth: ${ ({ separator }) => separator ? 6 : 1 };
  borderBottomColor: ${ ({ separator, theme }) => separator ? theme.colors.background : theme.colors.border };
`

const Label = styled(Text) `
  font-family: Lato;
`

export default ListMenuItem