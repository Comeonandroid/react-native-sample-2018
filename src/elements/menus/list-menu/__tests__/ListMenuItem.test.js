import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import { itBehavesLike } from '../../../../specs/shared_examples/index'

import ListMenuItem from '../ListMenuItem'

const setup = propOverrides => {
  const props = Object.assign({
    children: 'Test',
    separator: false,
    name: 'Name',
    onPress: jest.fn(),
  }, propOverrides)

  const component = <ListMenuItem {...props} />
  const wrapper = shallow(component)

  return {
    props,
    component,
    wrapper
  }
}

describe('<ListMenuItem />', () => {
  const { component } = setup()
  itBehavesLike('aThemedComponent', { subject: component })

  describe('separator: true', () => {
    const { component } = setup()
    itBehavesLike('aThemedComponent', { subject: component })
  })

  describe('onPress return name props', () => {
    const { props, wrapper } = setup()
    itBehavesLike('aThemedComponent', { subject: component })
    wrapper.simulate('press')
    expect(props.onPress).toHaveBeenCalledWith(props.name)
  })
})
