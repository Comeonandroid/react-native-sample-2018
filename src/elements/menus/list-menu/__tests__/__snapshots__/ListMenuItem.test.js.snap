// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListMenuItem /> dark: renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {},
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    className={undefined}
    separator={false}
    style={
      Array [
        Object {},
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#807c7c",
            "borderBottomColor": "#fcfcfc",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "paddingBottom": 0,
            "paddingLeft": 16,
            "paddingRight": 20,
            "paddingTop": 0,
          },
          null,
        ],
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fcfcfc",
            "fontFamily": "Open Sans",
          },
          Array [
            Object {
              "fontFamily": "Lato",
            },
            undefined,
          ],
        ]
      }
    />
    Test
  </View>
</View>
`;

exports[`<ListMenuItem /> light: renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {},
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    className={undefined}
    separator={false}
    style={
      Array [
        Object {},
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#eae8e0",
            "borderBottomColor": "#4b4c4d",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "paddingBottom": 0,
            "paddingLeft": 16,
            "paddingRight": 20,
            "paddingTop": 0,
          },
          null,
        ],
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Open Sans",
          },
          Array [
            Object {
              "fontFamily": "Lato",
            },
            undefined,
          ],
        ]
      }
    />
    Test
  </View>
</View>
`;

exports[`<ListMenuItem /> onPress return name props dark: renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {},
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    className={undefined}
    separator={false}
    style={
      Array [
        Object {},
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#807c7c",
            "borderBottomColor": "#fcfcfc",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "paddingBottom": 0,
            "paddingLeft": 16,
            "paddingRight": 20,
            "paddingTop": 0,
          },
          null,
        ],
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fcfcfc",
            "fontFamily": "Open Sans",
          },
          Array [
            Object {
              "fontFamily": "Lato",
            },
            undefined,
          ],
        ]
      }
    />
    Test
  </View>
</View>
`;

exports[`<ListMenuItem /> onPress return name props light: renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {},
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    className={undefined}
    separator={false}
    style={
      Array [
        Object {},
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#eae8e0",
            "borderBottomColor": "#4b4c4d",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "paddingBottom": 0,
            "paddingLeft": 16,
            "paddingRight": 20,
            "paddingTop": 0,
          },
          null,
        ],
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Open Sans",
          },
          Array [
            Object {
              "fontFamily": "Lato",
            },
            undefined,
          ],
        ]
      }
    />
    Test
  </View>
</View>
`;

exports[`<ListMenuItem /> separator: true dark: renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {},
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    className={undefined}
    separator={false}
    style={
      Array [
        Object {},
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#807c7c",
            "borderBottomColor": "#fcfcfc",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "paddingBottom": 0,
            "paddingLeft": 16,
            "paddingRight": 20,
            "paddingTop": 0,
          },
          null,
        ],
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fcfcfc",
            "fontFamily": "Open Sans",
          },
          Array [
            Object {
              "fontFamily": "Lato",
            },
            undefined,
          ],
        ]
      }
    />
    Test
  </View>
</View>
`;

exports[`<ListMenuItem /> separator: true light: renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {},
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    className={undefined}
    separator={false}
    style={
      Array [
        Object {},
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#eae8e0",
            "borderBottomColor": "#4b4c4d",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "paddingBottom": 0,
            "paddingLeft": 16,
            "paddingRight": 20,
            "paddingTop": 0,
          },
          null,
        ],
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "Open Sans",
          },
          Array [
            Object {
              "fontFamily": "Lato",
            },
            undefined,
          ],
        ]
      }
    />
    Test
  </View>
</View>
`;
